frontend/public/components/search.tsx

```diff
@@ -0,0 +1,284 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+
+import { AsyncComponent } from './utils/async';
+import { connectToModel } from '../kinds';
+import { DefaultPage } from './default-resource';
+import { requirementFromString } from '../module/k8s/selector-requirement';
+import { ResourceListDropdown } from './resource-dropdown';
+import { resourceListPages } from './resource-pages';
+import { withStartGuide } from './start-guide';
+import { split, selectorFromString } from '../module/k8s/selector';
+import { referenceForModel /*,kindForReference*/ } from '../module/k8s';
+import { history, LoadingBox, PageHeading, ResourceIcon } from './utils';
+import { Button, Chip, ChipGroup, ChipGroupToolbarItem, Expandable } from '@patternfly/react-core';
+import { SearchFilterDropdown } from './search-filter-dropdown';
+import { CloseIcon } from '@patternfly/react-icons';
+
+const ResourceList = connectToModel(({ kindObj, mock, namespace, selector, nameFilter }) => {
+  if (!kindObj) {
+    return <LoadingBox />;
+  }
+
+  const componentLoader = resourceListPages.get(referenceForModel(kindObj), () =>
+    Promise.resolve(DefaultPage),
+  );
+  const ns = kindObj.namespaced ? namespace : undefined;
+
+  return (
+    <AsyncComponent
+      loader={componentLoader}
+      namespace={ns}
+      selector={selector}
+      nameFilter={nameFilter}
+      kind={kindObj.crd ? referenceForModel(kindObj) : kindObj.kind}
+      showTitle={false}
+      showTextFilter={false}
+      autoFocus={false}
+      mock={mock}
+    />
+  );
+});
+
+const updateUrlParams = (k, v) => {
+  const url = new URL(String(window.location));
+  const sp = new URLSearchParams(window.location.search);
+  sp.set(k, v);
+  history.push(`${url.pathname}?${sp.toString()}${url.hash}`);
+};
+const getResourceName = (v: string) => {
+  return v.split('~').pop();
+};
+
+const SearchPage_: React.FC<SearchProps> = (props) => {
+  const [selectedItems, setSelectedItems] = React.useState(['']);
+  const [collaspedState, setCollaspedState] = React.useState([]);
+  const [selector, setSelector] = React.useState(selectorFromString(''));
+  const [searchParameters, setSearchParameters] = React.useState<SearchParameters>({
```

Comment:
You should have different state values for name and label. It's an antipattern to combine multiple values in an object with `useState`